tests/unit/lcos/async_cb_colocated.cpp

```diff
@@ -129,55 +129,55 @@ void test_remote_async_cb_colocated(test_client const& target)
-        hpx::id_type dec = dec_f.get();
-
-        callback_called.store(0);
-        hpx::future<boost::int32_t> f1 =
-            hpx::async_cb<call_action>(dec, &cb, 42);
-        HPX_TEST_EQ(f1.get(), 41);
-        HPX_TEST_EQ(callback_called.load(), 1);
-
-        callback_called.store(0);
-        hpx::future<boost::int32_t> f2 =
-            hpx::async_cb<call_action>(hpx::launch::all, dec, &cb, 42);
-        HPX_TEST_EQ(f2.get(), 41);
-        HPX_TEST_EQ(callback_called.load(), 1);
+//         hpx::id_type dec = dec_f.get();
+//
+//         call_action call;
+//
+//         callback_called.store(0);
+//         hpx::future<boost::int32_t> f1 = hpx::async_cb(call, dec, &cb, 42);
+//         HPX_TEST_EQ(f1.get(), 41);
+//         HPX_TEST_EQ(callback_called.load(), 1);
+//
+//         callback_called.store(0);
+//         hpx::future<boost::int32_t> f2 =
+//             hpx::async_cb(hpx::launch::all, call, dec, &cb, 42);
+//         HPX_TEST_EQ(f2.get(), 41);
+//         HPX_TEST_EQ(callback_called.load(), 1);
     }
 
-    {
-        increment_with_future_action inc;
-        hpx::shared_future<boost::int32_t> f =
-            hpx::async(hpx::launch::deferred, hpx::util::bind(&increment, 42));
-
-        callback_called.store(0);
-        hpx::future<boost::int32_t> f1 = hpx::async_cb(
-            inc, hpx::colocated(target), &cb, f);
-        hpx::future<boost::int32_t> f2 = hpx::async_cb(
-            hpx::launch::all, inc, hpx::colocated(target), &cb, f);
-
-        HPX_TEST_EQ(f1.get(), 44);
-        HPX_TEST_EQ(f2.get(), 44);
-        HPX_TEST_EQ(callback_called.load(), 2);
-    }
+//     {
+//         hpx::future<hpx::id_type> dec_f =
+//             hpx::components::new_<decrement_server>(hpx::colocated(target));
+//         hpx::id_type dec = dec_f.get();
+//
+//         callback_called.store(0);
+//         hpx::future<boost::int32_t> f1 =
+//             hpx::async_cb<call_action>(dec, &cb, 42);
+//         HPX_TEST_EQ(f1.get(), 41);
+//         HPX_TEST_EQ(callback_called.load(), 1);
+//
+//         callback_called.store(0);
+//         hpx::future<boost::int32_t> f2 =
+//             hpx::async_cb<call_action>(hpx::launch::all, dec, &cb, 42);
+//         HPX_TEST_EQ(f2.get(), 41);
+//         HPX_TEST_EQ(callback_called.load(), 1);
+//     }
+//
+//     {
+//         increment_with_future_action inc;
+//         hpx::shared_future<boost::int32_t> f =
+//             hpx::async(hpx::launch::deferred, hpx::util::bind(&increment, 42));
+//
+//         callback_called.store(0);
+//         hpx::future<boost::int32_t> f1 = hpx::async_cb(
+//             inc, hpx::colocated(target), &cb, f);
+//         hpx::future<boost::int32_t> f2 = hpx::async_cb(
+//             hpx::launch::all, inc, hpx::colocated(target), &cb, f);
+//
+//         HPX_TEST_EQ(f1.get(), 44);
+//         HPX_TEST_EQ(f2.get(), 44);
+//         HPX_TEST_EQ(callback_called.load(), 2);
+//     }
```

Comment:
On 05/03/2016 10:39 PM, Hartmut Kaiser wrote:
> In tests/unit/lcos/async_cb_colocated.cpp
> <https://github.com/STEllAR-GROUP/hpx/pull/2139#discussion_r61952000>:
>
>> +//
>> +//     {
>> +//         increment_with_future_action inc;
>> +//         hpx::shared_future<boost::int32_t> f =
>> +//             hpx::async(hpx::launch::deferred, hpx::util::bind(&increment, 42));
>> +//
>> +//         callback_called.store(0);
>> +//         hpx::future<boost::int32_t> f1 = hpx::async_cb(
>> +//             inc, hpx::colocated(target), &cb, f);
>> +//         hpx::future<boost::int32_t> f2 = hpx::async_cb(
>> +//             hpx::launch::all, inc, hpx::colocated(target), &cb, f);
>> +//
>> +//         HPX_TEST_EQ(f1.get(), 44);
>> +//         HPX_TEST_EQ(f2.get(), 44);
>> +//         HPX_TEST_EQ(callback_called.load(), 2);
>> +//     }
>
> This commented code is a leftover, I believe. Please re-enable the tests.

Thanks, should be fixed, finally...